server/src/main/java/org/cloudfoundry/identity/uaa/health/HealthzEndpoint.java

```diff
@@ -45,8 +54,28 @@ public String getHealthz(HttpServletResponse response) {
             response.setStatus(503);
             return "stopping\n";
         } else {
-            return "ok\n";
+            if (wasLastConnectionSuccessful == null) {
+                return "No Check done.\n";
+            }
+
+            if (wasLastConnectionSuccessful) {
+                return "ok\n";
+            } else {
+                response.setStatus(503);
+                return "Database Connection failed.\n";
+            }
         }
     }
 
-}
\ No newline at end of file
+    @Scheduled(fixedRateString = "${uaa.health.db.rate:10000}")
```

Comment:
Do we know if the schedule will run this right at startup? I'm just curious whether we're likely to get "No Check done" in the 10 seconds after startup.

Reply:
From my experience with this, the first run is done directly after the app is running (if there is no configuration of the "initialDelay" in the annotation. So the time where you would be able to see the No Check done should be rather small.